neqo-transport/tests/network.rs

```diff
@@ -148,6 +149,147 @@ simulate!(
     ],
 );
 
+// bandwidth 10Mbps, 20Mbps, 50Mbps and 100Mbps
+// delay by 10ms, 50ms, 100ms and 250ms
+// loss 1%, 2%, 5%, 10% and 20%
+// With tail-drop queue
+#[test]
+fn diff_conditions_client_download() {
+    for b in &[10, 20, 50, 100] {
+        for delay in &[10, 25, 50, 100, 250] {
+            for loss in &[0, 1, 2, 5, 10, 20] {
+                Simulator::new(
+                    format!("diff conditions download {} {} {}", b, *delay, *loss),
+                    boxed![
+                        ConnectionNode::new_client(boxed![
+                            ReachState::new(State::Confirmed),
+                            ReceiveData::new(TRANSFER_AMOUNT),
+                            ReachState::new(State::Closed(ConnectionError::Transport(
+                                Error::IdleTimeout
+                            )))
+                        ]),
+                        TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                        ConnectionNode::new_server(boxed![
+                            ReachState::new(State::Confirmed),
+                            SendData::new(TRANSFER_AMOUNT),
+                            ReachState::new(State::Closed(ConnectionError::Transport(
+                                Error::IdleTimeout
+                            )))
+                        ]),
+                        Drop::percentage(*loss),
+                        TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                    ],
+                )
+                .run();
+            }
+        }
+    }
+}
+
+// bandwidth 10Mbps, 20Mbps, 50Mbps and 100Mbps
+// delay by 10ms, 50ms, 100ms and 250ms
+// loss 1%, 2%, 5%, 10% and 20%
+#[test]
+fn diff_conditions_client_upload() {
+    for b in &[10, 20, 50, 100] {
+        for delay in &[10, 25, 50, 100, 250] {
+            for loss in &[0, 1, 2, 5, 10, 20] {
```

Comment:
Combinatorial explosion of 120 transfers.  Ouch.  I think that we might want to take these tests and put them in a new file behind a feature flag so that regular test runs don't run the whole matrix.

As far as I can see, this is identical to the later tests without loss, so maybe just drop those latter ones.